### ISSUE TITLE
Fails to create new task on OSX

### ISSUE THREAD

**Issue by jerearista** [puppetlabs/pdk#316]
Prior to this, I used PDK to create a new module, added PDK to the Gemfile in the new module then ran `bundle install` from the module directory.
```
$ bundle exec pdk new task hello_world
bundler: failed to load command: pdk (/Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/bin/pdk)
NameError: uninitialized constant Puppet::Util::Platform
Did you mean?  RUBY_PLATFORM
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/puppet-5.3.2-universal-darwin/lib/puppet/util/windows.rb:15:in `<module:Windows>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/puppet-5.3.2-universal-darwin/lib/puppet/util/windows.rb:1:in `<top (required)>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/util/windows.rb:6:in `require'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/util/windows.rb:6:in `<top (required)>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/util.rb:5:in `require'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/util.rb:5:in `<top (required)>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/cli/exec_group.rb:4:in `require'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/cli/exec_group.rb:4:in `<top (required)>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/cli.rb:8:in `require'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/lib/pdk/cli.rb:8:in `<top (required)>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/exe/pdk:3:in `require'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/gems/pdk-1.2.0/exe/pdk:3:in `<top (required)>'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/bin/pdk:23:in `load'
  /Users/jere/src/bolt/pdk_mod/.bundle/gems/ruby/2.3.0/bin/pdk:23:in `<top (required)>'
```

**Comment by DavidS** [comment]
You should never add the pdk to the Gemfile of the modules you are working on. If you are using the native package, you just run `pdk new task` after changing into the module's directory, without having to do any additional work. If you can't use the native package, please tell us why, and install the gem into your system ruby until we can sort that out for you.

**Comment by LongLiveCHIEF** [comment]
I get this even when installing the gem into the system ruby, using the official Ruby:2.4 docker image

**Comment by jerearista** [comment]
For now, just use the [installer package](https://puppet.com/download-puppet-development-kit) which includes a self-contained Ruby and dependencies.

**Comment by LongLiveCHIEF** [comment]
Not an option.

**Comment by jerearista** [comment]
What if you disassembled the .deb or .rpm package and (rpm2cpio) and placed the files in your image?

**Comment by mrmarbury** [comment]
Hi,

I am developing puppet on a FreeBSD host. Since there is no pdk package for FreeBSD, I can only use pdk by installing it as a gem. And I get the same error as the OP. 
Puppet development and testing works as intended without pdk. So I assume pdk is doing something weird here?

Thx for the support,
Stefan

**Comment by LongLiveCHIEF** [comment]
I'll give it a shot and let you know.

**Comment by cliffano** [comment]
> For now, just use the [installer package](https://puppet.com/download-puppet-development-kit) which includes a self-contained Ruby and dependencies.

Is this still the same case in second half of 2018?

**Comment by DavidS** [comment]
@cliffano this is fixed on master and will be in the upcoming 1.7.1 release. See https://tickets.puppetlabs.com/browse/PDK-1099

**Comment by cliffano** [comment]
@DavidS thanks for the pointer. I tried master and I'll wait until 1.8.0 is released, will resort to pre-packaged pdk for now.
